### PR TITLE
fix clusterset sync issue

### PR DESCRIPTION
**Pull request by elgnay**
Signed-off-by: Yang Le <yangle@redhat.com>

### CODE REVIEW

File: pkg/hub/managedclusterset/controller.go [hunk 1]
```diff
@@ -177,46 +148,31 @@ func (c *managedClusterSetController) syncClusterSet(ctx context.Context, origin
...
-// given cluster set
-func (c *managedClusterSetController) updateClusterSetsMap(clusterSetName string, clusters []*clusterv1.ManagedCluster) {
-	c.mapLock.Lock()
-	defer c.mapLock.Unlock()
-
-	newMembers := sets.NewString()
-	for _, cluster := range clusters {
-		newMembers.Insert(cluster.Name)
+func getClusterSet(obj interface{}) string {
+	accessor, _ := meta.Accessor(obj)
```
Comment (qiujian16) [inline pkg/hub/managedclusterset/controller.go]: you would need to handle tombstone case

**Review by qiujian16 — COMMENTED** [review]
I think we should add an integration test for this

File: pkg/hub/managedclusterset/controller.go (hunk 1 above)
Comment (elgnay) [reply pkg/hub/managedclusterset/controller.go]: Done.

**Comment by elgnay** [comment]
> I think we should add an integration test for this

We have already had the integration test case to cover this issue, but the out-of-sync happens occasionally. I update the test case to repeat the same testing 10 times.

**Pull request merged**